rust/src/treefile.rs

```diff
@@ -445,6 +448,97 @@ impl Treefile {
         }))
     }
 
+    /// Return the raw file descriptor for the workdir
+    pub(crate) fn get_dfd(&self) -> i32 {
+        self.primary_dfd.as_raw_fd()
+    }
+
+    /// Return the raw file descriptor for the postprocess script
+    pub(crate) fn get_postprocess_script_fd(&mut self) -> i32 {
+        self
+            .externals
+            .postprocess_script
+            .as_mut()
+            .map_or(-1, raw_seeked_fd)
```

Comment:
I'm mildly worried about this `-1` as it blurs the line between a `RawFd` and a `Option<RawFd>`.
Should we add a `type OptRawFd = i32` to remember to undo this?

Reply:
Yes...actually I think we're way overdue for a bridged `FileDescriptor` type that's bridged - something like an `Rc<i32>` and we'd also have `OptionalFileDescriptor` or so.  In some cases for example when we convert a fd to a `/proc/self/fd` path (mostly for libdnf/librpm) we lose "ownership" tracking so we'd want to use `g_object_set_data()` or so on the `DnfContext` so that we'd ref/unref the fd when the context goes away.

Reply:
That said unfortunately cxx-rs doesn't support type aliases yet.  I'll try to do a more coherent approach to file descriptors as a followup?

Reply:
Ah wops yes, I was assuming type aliases where a low-resistance path for both world. Let's defer this to a later cleanup.